sync.js

```diff
@@ -463,13 +464,21 @@ function WifiPeer (connection, info) {
 // key is String or Buffer
 function hash (key) {
   if (typeof key === 'string') {
-    key = Buffer.from(key, 'hex')
-  }
-  if (Buffer.isBuffer(key) && key.length === 32) {
-    return crypto.discoveryKey(key).toString('hex')
-  } else {
-    return null
+    key = Buffer.from(key)
   }
+  if (!Buffer.isBuffer(key) || key.length === 0) return null
```

Comment:
Heck, how about we just make it `throw`? That way nobody will get burned like you did again.